libraries/AC_PID/AC_PID_2D.cpp

```diff
@@ -133,16 +133,17 @@ void AC_PID_2D::update_i(const Vector2f &limit)
     _pid_info_y.limit = false;
 
     Vector2f delta_integrator = (_error * _ki) * _dt;
+    float integrator_length = _integrator.length();
```

Comment:
could be inside the if() ?

Reply:
Done

Reply:
No, we need to calculate this before we change that vector.